boards/arm: Adding support for nRF9160 InnBlue21 Board.

olsky opened the pull request:
Adding support for nRF9160 based InnBlue board V2.1. Supports both
Secure and Non-Secure configurations along with various sensors
(lis2dh12 / hts221) and devices(i2c / pwm).

Signed-off-by: Oleh Lozynskyy <olsky@users.noreply.github.com>

carlescufi reviewed (approved):
LGTM, no objections

carlescufi reviewed (changes requested):
Hold on, I didn't realize this is missing the corresponding documentation.

@olsky please add a `doc/index.rst` similar to https://github.com/zephyrproject-rtos/zephyr/tree/master/boards/arm/nrf9160_pca10090/doc/index.rst

carlescufi commented:
>Signed-off-by: Oleh Lozynskyy olsky@users.noreply.github.com

Please use a valid email address

ioannisg reviewed (commented):
looks fine to me - thanks for the contributions
Besides the issues mentioned by @carlescufi , please, fix, at least, the one check-patch warning with the line exceeding 80 chars, if possible.
Also, please check if you can resize the added board image - looks over-sized to me (but maybe i am wrong).

olsky commented:
> ease check if you can resize the added board ima

@ioannisg sure, let me fix that...

mbolivar-nordic reviewed (dismissed):
We need a real email address, not `Signed-off-by: Oleh Lozynskyy <olsky@users.noreply.github.com>` type addresses. Any commits with the noreply emails were merged by mistake.

bwasim committed:
boards/arm: Adding support for nRF9160 InnBlue21 Board.

Adding support for nRF9160 based InnBlue board V2.1. Supports both
Secure and Non-Secure configurations along with various sensors
(lis2dh12 / hts221) and devices(i2c / pwm).

Signed-off-by: Bilal Wasim <bilalwasim676@gmail.com>

bwasim commented:
> Besides the issues mentioned by @carlescufi , please, fix, at least, the one check-patch warning with the line exceeding 80 chars, if possible.

@ioannisg , I would've fixed this but didn't because it was a URL. The same problem exists with the nRF9160 DK board support as well. Don't think its possible to fix.

bwasim commented:
@reviewers, All requested changes are addressed in the pull request. Please review..

carlescufi commented:
@olsky please squash all commits together and then force-push

carlescufi reviewed (approved):
OK with this once the commits are squashed

carlescufi commented:
@ioannisg and @mbolivar-nordic can you re-review?

mbolivar-nordic commented on boards/arm/nrf9160_innblue21/Kconfig.defconfig:
```diff
@@ -0,0 +1,50 @@
+config BOARD
+	default "nrf9160_innblue21"
+
+config I2C_2
```
I believe this is no longer needed since bf0c4a841d was merged. @anangl ?

mbolivar-nordic commented on boards/arm/nrf9160_innblue21/doc/index.rst:
```diff
@@ -0,0 +1,105 @@
+.. _nrf9160_innblue21:
+
+nRF9160-INNBLUE21
```
Various comments I made in the PR for the other board (https://github.com/zephyrproject-rtos/zephyr/pull/23451#pullrequestreview-386044061) also apply to the documentation in this PR.

Can you please take a look there and let's resolve those issues, then apply any necessary changes here too?

bwasim replied:
Hi @mbolivar-nordic / @ioannisg , I've made the required changes here and in the other board (https://github.com/zephyrproject-rtos/zephyr/pull/23451) and squashed commits as well.. 

Can you recheck and let me know if there is anything else we need to add to the board defs ? Thanks..

mbolivar-nordic replied:
Thanks!

Can you please clean up the commit log? There's no need to include the changes made in response to each round of review in the commit log. Just describe the changes made by the commit.

I know the git squash UI opens your editor with the commit messages concatenated, but the idea is that you then edit them together into a single commit message.

bwasim replied:
Done.. There was also another problem in the defconfig where we were modifying an non-selectable symbol (NRFX_UART0) which was causing shippable to complain.. I've fixed that and hopefully we should be good to go here..

mbolivar-nordic commented on boards/arm/nrf9160_innblue21/doc/index.rst:
```diff
@@ -0,0 +1,105 @@
+The "innblue v2.1" is a cellular IoT sensor development board, which
+is based on nRF9160 SiP featuring NB-IoT and LTE-M connectivity.
+
+.. figure:: img/nrf9160_innblue21.png
```
This looks like a very nice board. I may buy one myself :).

olsky replied:
thanks! indeed, nice :) @mbolivar-nordic if you are in Trondheim few boards should be there :)

mbolivar-nordic replied:
I am not. Are they commercially available? I can't tell from the docs or innblue website.

bwasim commented:
@all, please re-review.. All tests are passing now..

mbolivar-nordic commented on boards/arm/nrf9160_innblue21/doc/index.rst:
```diff
@@ -0,0 +1,145 @@
+Overview
+********
+
+The "nRF9160_innblue21" is a cellular IoT sensor development board, which
```
```suggestion
The nRF9160 innblue21 is a cellular IoT sensor development board, which
```

This is the same suggestion made for the 22 variant.

mbolivar-nordic reviewed (approved):
Docs look fine. One suggestion to consider. Thanks for your patience and contribution.

olsky commented:
@mbolivar-nordic thank you! COmmercially -- not yet, we have some homework todo... we will notify...
Best
Oleh

ioannisg commented on boards/arm/nrf9160_innblue21/doc/index.rst:
```diff
@@ -0,0 +1,145 @@
+* CryptoElement: Atmel ATECC608a
+* Humidity Sensor: ST HTS221
+* Qi charger: TI BQ51013
+* Battery guel gauge: TI BQ27421
```
fuel* ??

ioannisg reviewed (approved):
Looks good now, I had a minor comment about a possible typo

@olsky ^^

olsky commented:
> Looks good now, I had a minor comment about a possible typo
> 
> @olsky ^^

sorry for a bit late reply, but these weeks are crazy and to keep with time and schedule is a real challenge (adding the limit of movement) :-)

thank you for your time and reviews! :)

ioannisg commented:
> > Looks good now, I had a minor comment about a possible typo
> > @olsky ^^
> 
> sorry for a bit late reply, but these weeks are crazy and to keep with time and schedule is a real challenge (adding the limit of movement) :-)
> 
> thank you for your time and reviews! :)

All what seems to be left is a question about a possible typo :)

ioannisg commented:
@olsky the build-errors are relevant (I believe, due to recent DT changes)

henrikbrixandersen commented:
Duplicate efforts: #23451

mbolivar-nordic commented:
> Duplicate efforts? #23451

21 != 22

olsky commented:
> > Duplicate efforts? #23451
> 
> 21 != 22

yes, thanks!
that's true: these are different boards, we were not creative with names :)
so the base is similar, but content (parts and sensors) and usage are different.

bwasim commented:
@carlescufi , can we merge this also ? Let me know if more updates are required..
